### PR TITLE
Révision importante du processus de traitement des données de prélèvements

### PR DESCRIPTION
**Pull request by jdesboeufs**
This pull request introduces several significant improvements related to OpenAPI specification management, cron job configurability, and data processing for attachments and series. The changes modernize API documentation practices, add automated linting, enhance the cron job system for more flexible deployments, and refactor the attachment processing pipeline for better performance and maintainability.

The most important changes are:

**OpenAPI Specification & Linting**
- Migrated API documentation to a single `docs/openapi.yaml` OpenAPI file, removing the static route table from `README.md` and providing instructions for using and linting the spec. Added a Spectral configuration (`.spectral.yaml`) and a GitHub Actions workflow to automatically lint OpenAPI changes on pull requests. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L98-R124) [[2]](diffhunk://#diff-d5da7cb43c444434994b76f3b04aa6e702c09e938de09dbc09d72569d611d9abR1-R9) [[3]](diffhunk://#diff-2ea7fe5ab9c07a9fa8bddcecbf529dab8252d8a2de05632e3475ae6930559b67R1-R32)

**Cron Job System Improvements**
- Refactored the cron job system in `lib/cron.js` to allow selection of specific cron jobs via the `CRON_JOBS` environment variable. Jobs now have explicit names, and unknown job names are warned about at startup. The `.env.example` file documents the new variable. [[1]](diffhunk://#diff-ad1181afb313b7b611e9e10ffa756b5398b08da6314e5ee0c424b08650eb2e38R1-R2) [[2]](diffhunk://#diff-ad1181afb313b7b611e9e10ffa756b5398b08da6314e5ee0c424b08650eb2e38R11-R55) [[3]](diffhunk://#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR21-R23)

**Attachment and Series Processing Refactor**
- Major refactor of `lib/demarches-simplifiees/attachments.js`:
  - Added modular parsing, error normalization, and hashing utilities for series.
  - Implemented logic to compare new and existing series by content hash, only inserting/deleting as needed.
  - Enhanced error reporting and result summaries for attachment processing.
  - Added total volume calculation for series. [[1]](diffhunk://#diff-0c8b8dda6926a313ba65223635add90b871a489bfaee5e2ba6b8ee6113dcc122R23-R131) [[2]](diffhunk://#diff-0c8b8dda6926a313ba65223635add90b871a489bfaee5e2ba6b8ee6113dcc122L50-R240) [[3]](diffhunk://#diff-0c8b8dda6926a313ba65223635add90b871a489bfaee5e2ba6b8ee6113dcc122L31-R144)

**Testing and Validation**
- Added comprehensive AVA tests for day comparison logic in `lib/demarches-simplifiees/__tests__/consolidate.js`, covering edge cases and dossier status transitions.

**Authorization & Indexing**
- Introduced `checkPermissionOnSeries` middleware for series-level authorization in `lib/auth/middleware.js`.
- Ensured MongoDB indexes for series are created at application startup in `api.js`.

These changes improve maintainability, deployment flexibility, and data integrity across the application.

### CODE REVIEW

File: lib/demarches-simplifiees/attachments.js [hunk 1]
```diff
@@ -47,53 +156,88 @@ async function processAttachment(attachment) {
...
+
+    for (const dataPoint of s.data) {
+      if (typeof dataPoint.value === 'number') {
+        total += dataPoint.value
+      }
+    }
+  }
+
+  return total
+}
```
[the remark this answers is not in the document]
Comment (jdesboeufs) [reply lib/demarches-simplifiees/attachments.js]: Pas nécessaire.

File: lib/models/series.js [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+  attachmentId,
+  pointId,
+  preleveurId,
+  from,
+  to,
+  onlyIntegratedDays = false
+}) {
+  if (!territoire) {
+    throw new Error('territoire est obligatoire pour listSeries')
+  }
```
[the remark this answers is not in the document]
Comment (jdesboeufs) [reply lib/models/series.js]: C'est le cas un peu plus bas
